crates/rslint_parser/src/syntax/stmt.rs

```diff
@@ -549,13 +550,18 @@ pub(super) fn parse_block_impl(
 	Present(m.complete(p, block_kind))
 }
 
-#[must_use]
-pub(crate) fn directives(p: &mut Parser) -> Option<ParserState> {
-	let list = p.start();
+struct DirectivesList {
+	old_state: Option<ParserState>,
+}
 
-	let mut old_state: Option<ParserState> = None;
+impl Default for DirectivesList {
+	fn default() -> Self {
+		Self { old_state: None }
+	}
+}
 
-	fn is_directive(p: &Parser) -> bool {
+impl DirectivesList {
+	pub fn at_directives(&mut self, p: &mut Parser) -> bool {
```

Comment:
Can we make this private? 
```suggestion
	fn is_at_directive(&mut self, p: &mut Parser) -> bool {
```